test/NNPDE_tests.jl

```diff
@@ -553,3 +553,32 @@ p_ = res.minimizer[end-2:end]
 # u_predict  = [[discretization.phi[i]([t],minimizers[i])[1] for t in ts] for i in 1:3]
 # plot(sol)
 # plot!(ts, u_predict, label = ["x(t)" "y(t)" "z(t)"])
+
+@parameters t
+@variables i(..)
+Di = Differential(t)
+Ii = Integral(t, DomainSets.ClosedInterval(0, t))
+eq = Di(i(t)) + 2*i(t) + 5*Ii(i(t)) ~ 1
+bcs = [i(0.) ~ 0.0]
+domains = [t ∈ Interval(0.0,2.0)]
+chain = FastChain(FastDense(1,15,Flux.σ),FastDense(15,1))
+initθ = Float64.(DiffEqFlux.initial_params(chain))
+strategy_ = NeuralPDE.GridTraining(0.05)
+discretization = NeuralPDE.PhysicsInformedNN(chain,
+                                             strategy_;
+                                             init_params = nothing,
+                                             phi = nothing,
+                                             derivative = nothing,
+                                             )
+pde_system = PDESystem(eq,bcs,domains,[t],[i])
+prob = NeuralPDE.discretize(pde_system,discretization)
```

Comment:
it work in forward but
```julia
prob.f.f.loss_function(initθ)
1.1225828958232524
```
donesn't work in backward
```julia
julia> Zygote.gradient(prob.f.f.loss_function,initθ)
ERROR: MethodError: no method matching get(::Tuple{Int64, Int64}, ::Int64, ::Int64)
```
I suppose it is related to integral function.
So you need  that work in both way:
`integral = get_numeric_integral(strategy, _indvars, _depvars, dict_indvars, chain, derivative)`

```julia
integral(initθ,... )
integral_ = (θ) -> integral(θ, ...)
Zygote.gradient(integral_,initθ)
```
All symbolic things should do before calculation because `Zygote.gradient` probably can't handle all of it.